eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventData.java

```diff
@@ -48,8 +50,19 @@
  * </p>
  */
 public class EventData implements Comparable<EventData> {
-    private final Map<String, Object> properties = new HashMap<>();
-    private transient ByteBuffer data;
+    /*
+     * These are properties owned by the service and set when a message is received.
+     */
+    static final Set<String> RESERVED_SYSTEM_PROPERTIES = Collections.unmodifiableSet(new HashSet<String>() {{
+            add(OFFSET_ANNOTATION_NAME.getValue());
+            add(PARTITION_KEY_ANNOTATION_NAME.getValue());
```

Comment:
Well, this just caused me to question Id versus Key and found an error in the .NET code as a result.   
![](https://media2.giphy.com/media/l4JyRqcDU93S334KQ/giphy.gif)